packages/pagination/src/index.ts

```diff
@@ -215,3 +218,72 @@ export const createPagination = (
 
   return [paginationProps, page, setPage as Setter<number>];
 };
+
+declare module "solid-js" {
+  namespace JSX {
+    interface Directives {
+      loaderDirective: boolean
+    }
+  }
+}
+
+export type _E = JSX.Element;
+
+/**
+ * ```typescript
+ * const [pages, loaderDirective, { page, setPage, setPages, end, setEnd }] = createInfiniteScroll(fetcher);
+ * ```
+ * @param fetcher (page: number) => Promise<T[]>
+ * @return `pages()` is an accessor contains array of contents 
+ * @property `pages.loading` is a boolean indicator for the loading state
+ * @property `pages.error` contains any error encountered
+ * @return `loaderDirective` is an directive used to set loader
+ * @method `page` is an accessor that contains page number
+ * @method `setPage` allows to manually change the page number
+ * @method `setPages` allows to manually change the contents of page
+ * @method `end` is a boolean indicator for end of page
+ * @method `setEnd` allows to manually change the end
+ *
+ */
+export function createInfiniteScroll<T>(
+  fetcher: (page: number) => Promise<T[]>,
+): [
+    pages: Accessor<T[]>,
+    loaderDirective: Setter<Element | undefined>,
```

Comment:
Please alias the type into LoaderDirective instead of using that as name:

```ts
type LoaderDirective = (ref: Element) => void;

return {
  …,
  asLoader: LoaderDirective,
  …
};
```

Reply:
i misread the message. i will fix it.